extensions/barrage/src/main/java/io/deephaven/extensions/barrage/chunk/VarBinaryChunkInputStreamGenerator.java

```diff
@@ -155,30 +300,33 @@ public void visitBuffers(final BufferListener listener) {
         @Override
         protected int getRawSize() {
             if (cachedSize == -1) {
-                cachedSize = 0;
+                MutableLong totalCachedSize = new MutableLong(0L);
                 if (sendValidityBuffer()) {
-                    cachedSize += getValidityMapSerializationSizeFor(subset.intSize(DEBUG_NAME));
+                    totalCachedSize.add(getValidityMapSerializationSizeFor(subset.intSize(DEBUG_NAME)));
                 }
 
                 // there are n+1 offsets; it is not assumed first offset is zero
-                if (!subset.isEmpty() && subset.size() == myOffsets.size() - 1) {
-                    cachedSize += myOffsets.size() * Integer.BYTES;
-                    cachedSize += myOffsets.get(subset.intSize(DEBUG_NAME)) - myOffsets.get(0);
+                if (!subset.isEmpty() && subset.size() == myByteStorage.offsets.size() - 1) {
+                    totalCachedSize.add(myByteStorage.offsets.size() * Integer.BYTES);
+                    for (int i = 0; i < myByteStorage.size(); i++) {
+                        totalCachedSize.add(myByteStorage.getByteArraySize(i));
+                    }
```

Comment:
Another benefit of changing byteStorage to think in overall offsets means that this calculation becomes O(1) again.